Orders.Logic/OrderManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Orders.Logic
+{
+    public class OrderManager
+    {
+        private readonly IOrderStore orderStore;
+        public OrderManager(IOrderStore orderStore)
+        {
+            this.orderStore = orderStore;
+        }
+        public void WriteOutSmallOrders()
+        {
+            var orders = orderStore.GetOrders();
+            SmallOrderFilter filter = new SmallOrderFilter(new OrderWriter(),
+           orders);
+            filter.WriteOutFiltrdAndPriceSortedOrders(new OrderWriter());
+        }
+        public void WriteOutLargeOrders()
+        {
+            var orders = orderStore.GetOrders();
+            LargeOrderFilter filter = new LargeOrderFilter(new OrderWriter(),
+           orders);
+            filter.WriteOutFiltrdAndPriceSortedOrders(new OrderWriter());
+        }
+    }
+    public class LargeOrderFilter
+    {
+        private IOrderWriter orderWriter;
+        private List<Order> orders;
+        public LargeOrderFilter(IOrderWriter orderWriter, List<Order> orders)
+        {
+            filterSize = "100";
+            this.orderWriter = orderWriter;
+            this.orders = orders;
+        }
+        protected string filterSize;
+        public void WriteOutFiltrdAndPriceSortedOrders(IOrderWriter writer)
+        {
+            List<Order> filteredOrders = this.FilterOrdersSmallerThan(orders,
+           filterSize);
+            Enumerable.OrderBy(filteredOrders, o => o.Price);
+            ObservableCollection<Order> observableCollection =
+            new ObservableCollection<Order>();
+            foreach (Order o in filteredOrders)
+            {
+                observableCollection.Add(o);
+            }
+            writer.WriteOrders(observableCollection);
+        }
+        protected List<Order> FilterOrdersSmallerThan(List<Order> allOrders,
+       string size)
+        {
+            List<Order> filtered = new List<Order>();
+            for (int i = 0; i <= allOrders.Count; i++)
+            {
+                int number = orders[i].toNumber(size);
+                if (allOrders[i].Size <= number)
+                {
+                    continue;
+                }
+                else
+                {
+                    filtered.Add(orders[i]);
+                }
+            }
+            return filtered;
+        }
+    }
+    public class SmallOrderFilter : LargeOrderFilter
+    {
```

Comment:
Order size does not warrant inheritance. If there was indeed a case for inheritance, it would be applied to the Orders with overriding  (and / or additional) functionality...